tap_eloqua/sync.py

```diff
@@ -377,12 +377,12 @@ def sync_activity_stream(client,
                          activity_type):
     finished = False
     sync_start = pendulum.now('UTC')
-    end_date = sync_start
+    end_date, last_sync_date = sync_start, None
```

Comment:
Are these changes dev-mode related? If not can we keep it separate from dev-mode changes.

Reply:
And if these are required changes then update the description on this PR.

Reply:
@RushiT0122 
These are not dev mode related changes, but if you see the code from Line:380 - Line:403
`last_sync_date` was declared on Line:385 in the try statement, but accessed outside its declared scope in the except block, hence it was introduced here.

Reply:
I'm pretty sure this is a problem in other languages, but not python

```python
>>> for i in range(3):
...   try:
...     print("About to initialize `my_variable`")
...     my_variable = i
...     print(f"my_variable: {my_variable}")
...     raise RuntimeError(f"Problem occured with {my_variable}")
...   except Exception as err:
...     print(f"Caught {type(err)} with message {str(err)}")
...     print(f"Current value of my_variable {my_variable}")
...
About to initialize `my_variable`
my_variable: 0
Caught <class 'RuntimeError'> with message Problem occured with 0
Current value of my_variable 0
About to initialize `my_variable`
my_variable: 1
Caught <class 'RuntimeError'> with message Problem occured with 1
Current value of my_variable 1
About to initialize `my_variable`
my_variable: 2
Caught <class 'RuntimeError'> with message Problem occured with 2
Current value of my_variable 2
```